drivers/sensor/bme680/Kconfig

```diff
@@ -0,0 +1,137 @@
+# Kconfig.bme680 - BME680 temperature, pressure, humidity and gas sensor configuration options
+
+#
+# Copyright (c) 2018 Bosch Sensortec GmbH
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+menuconfig BME680
+	bool "BME680 sensor"
+	depends on I2C
+	help
+	 Enable driver for BME680 I2C-based based temperature, pressure, humidity and gas sensor.
+
+if BME680
+
+config BME680_NAME
+	string "Device name"
+	default "BME680"
+
+config BME680_I2C_NAME
+	string "I2C device name"
+	default "I2C_0"
+
+config BME680_I2C_ADDRESS
```

Comment:
You can remove NAME, I2C_NAME, and I2C_ADDRESS from Kconfig now and rely exclusively on device tree for them. Please see #10115 for reference

Reply:
@MaureenHelm,

How does one configure the defaults in the dts configuration when using  `HAS_DTS_I2C`

Reply:
You'd do that either by using the board's defaults from DTS, or by creating an overlay in your application's folder and the build system will pick it up. Either way, the content will be pretty much the same, for example: 
```
&i2c2 {
	status = "ok";
	sda-pin = <11>;
	scl-pin = <12>;

	bme680@76 {
		compatible = "bosch,bme680";
		label = "BME680";
		reg = <0x76>;
		clock-frequency = <I2C_BITRATE_FAST>;
	};
};
```

Then the BME680's I2C device is found similar to this:
https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/sensor/bme280/bme280.c#L360

And the address like this:
https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/sensor/bme280/bme280.c#L367

Reply:
As @jtguggedal said, and then you can remove `BME680_NAME`, `BME680_I2C_NAME` and `BME680_I2C_ADDRESS` completely from this file, since the info will be coming from DT

Reply:
Hi @carlescufi,

Understood the changes here in the Kconfig files.  It is not clear how `DT_BOSCH_BME280_0_BUS_NAME` is defined in the source code.

Reply:
It is not defined in the source code, it will be auto-generated by the Device Tree preprocessing that Zephyr runs. If you build a sample that uses Device Tree and then take a look at:

`build/zephyr/include/generated/generated_dts_board_unfixed.h`

you should see those there

Reply:
Just to add to what @carlescufi wrote, these files are included when you have for example `kernel.h` or `device.h` included in your source file, which you already do in `bme680.c`. So just changing to the defines extracted from DTS should work.

Reply:
@carlescufi and @jtguggedal,

Thank you for the explanation. So what is the name(format) of the macro generated by the DTS pre-processing?

Reply:
@BoschSensortec The content of `build/zephyr/include/generated/generated_dts_board_unfixed.h` is generated from the .dts files and overlays. The format is `DT_<COMPAT>_<INSTANCE>_<PROP`, where `<COMPAT>` is the extracted from the `compatible` string in the dts node, `<INSTANCE>` is the instance number (0 in the case where only one), and `<PROP>` is the name of the property, for example `BUS_NAME`. Please corerct me if I'm wrong, @carlescufi @galak .

Reply:
So one would replace it with `DT_BOSCH_BME680_0_BASE_ADDRESS`, `DT_BOSCH_BME280_0_BUS_NAME` and `DT_BOSCH_BME280_0_NAME` in the sources?

Reply:
@BoschSensortec That's right, these will be generated from DTS automatically instead of from Kconfig now. You can use this commit converting the BME280 driver to use devie tree for reference:
https://github.com/zephyrproject-rtos/zephyr/commit/be2b6f870d869f736f35c8ed1eaaad51097bc0e1#diff-9c041a0490ec34108eb76ab1d632827a